format/Schema.fbs

```diff
@@ -187,6 +187,103 @@ table Interval {
   unit: IntervalUnit;
 }
 
+
+/// ----------------------------------------------------------------------
+/// user defined key value pairs to add custom metadata to arrow
+/// key namespacing is the responsibility of the user
+
+table KeyValue {
+  key: string;
+  value: string;
+}
+
+/// ------------------------------------------------------------------------
+/// Types used to represent a packed c-like structs.  These are experimental
+/// and maybe subject to change.
+/// IMPORTANT: COMPATIBILITY MAY BE BROKEN IN FUTURE release.
+/// They are not required to be implemented.
+
+/// Similar to a c-union, takes the byte-width
+/// of its longest sub-type.  To record type of union
+/// use the "tagged union" pattern.
+table PackedUnion {
```

Comment:
Not sure this should be in the first version of this spec.

Reply:
I'm ok leaving it out.   I'm also leaning toward leaving out the PackedBinary and PackedUtf fields since it is easier to add then remove in flatbuffers.